service/history/api/startworkflow/api.go

```diff
@@ -101,19 +101,16 @@ func NewStarter(
 	tokenSerializer common.TaskTokenSerializer,
 	visibilityManager manager.VisibilityManager,
 	request *historyservice.StartWorkflowExecutionRequest,
-) (*Starter, error) {
-	namespaceEntry, err := api.GetActiveNamespace(shardContext, namespace.ID(request.GetNamespaceId()))
-	if err != nil {
-		return nil, err
-	}
```

Comment:
I moved the namespace lookup out of here to simplify this function and made the caller responsible for passing in the namespace. It makes the use of the Starter simpler IMO.